tools/rosbag/src/recorder.cpp

```diff
@@ -124,6 +126,18 @@ Recorder::Recorder(RecorderOptions const& options) :
     split_count_(0),
     writing_enabled_(true)
 {
+    if(options_.custom_freq){
+        try {
+                custom_freq_config_ = YAML::LoadFile(options_.file_name);
+        } catch (std::exception& e) {
+                ROS_ERROR_STREAM("Error loading file " <<options_.file_name);
+                ROS_DEBUG_STREAM(e.what());
+                exit (EXIT_FAILURE);
+        }
+        for(YAML::const_iterator it = custom_freq_config_.begin(); it!=custom_freq_config_.end(); ++it){
+            custom_record_freq_.insert_or_assign(it->first.as<std::string>(), it->second.as<int>());
```

Comment:
I think it would be nicer to print a warning if there is a duplicate entry, instead of silently overwriting.